src/compiler/checker.ts

```diff
@@ -22351,8 +22351,11 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             // from the target union, across all members
             const properties = getPropertiesOfType(target);
             const numericNamesOnly = isTupleType(source) && isTupleType(target);
-            for (const targetProp of excludeProperties(properties, excludedProperties)) {
+            for (let targetProp of excludeProperties(properties, excludedProperties)) {
                 const name = targetProp.escapedName;
+                if (target.flags & TypeFlags.Intersection) {
+                    targetProp = getPropertyOfUnionOrIntersectionType(target as IntersectionType, name, /*skipObjectFunctionPropertyAugment*/ true) || targetProp;
+                }

```

Comment:
This is only needed for this extra properties check:
https://github.dev/microsoft/TypeScript/blob/f7e4f809acaea6b1b247ed05b2b7873d4546ab9e/src/compiler/checker.ts#L21338-L21349

But I'm also not sure if this is the right fix, nor if it's complete right now. I think it's important to establish what this should:
```ts
interface Test1 { toString: null | 'string'; }
type Test3 = Test1 & { optional?: unknown };

declare const test3: Test3
test3.toString // what this should be?
```

I'm not sure but perhaps this *shouldn't* be an intersection like `"string" & (() => string)`? Should this just be `null | 'string'`? If the latter then `skipObjectFunctionPropertyAugment` should be used on a different layer, when resolving the properties of an intersection type.

@RyanCavanaugh any thoughts about this? :)